src/beamlime/applications/_nexus_helpers.py

```diff
@@ -238,51 +319,38 @@ def _merge_message_into_nexus_group_store(
 
     Parameters
     ----------
-    nexus_group_store:
-        The data module store that holds the data.
     structure:
         The nexus structure.
-    data_piece:
+    nexus_store:
+        The module store that holds the data to be merged in the template.
+    data:
         The content of the message.
-    path_matching_func:
+    find_insert_paths:
         A function that returns the paths that match the message.
-    data_field_initialize_func:
-        A function that initializes the datasets in the module.
+    initialize:
+        A function that initializes the dataset / group.
         *Initialize is done only when the relevant message arrives.*
         It is because we should distinguish between empty dataset and
         unexpectedly-not-receiving data.
-        It also depends on the ``data_piece`` to handle optional fields.
-    merge_func:
-        A function that merges the message into the store.
+        It also depends on the ``data`` to handle optional fields.
+    merge_message:
+        A function that merges the message into the initialized nexus dataset / group.
 
     Side Effects
     ------------
     The ``nexus_group_store`` is updated with the merged data.
 
     """
-    for path in path_matching_func(structure, data_piece):
-        if path not in nexus_group_store:
-            parent = find_nexus_structure(structure, path[:-1])
-            # Validate the module place holder in the parent
-            if len(parent["children"]) > 1:
-                raise ValueError("Multiple modules found in the same data group.")
-            # Initialize the data module storage.
-            nexus_group_store[path] = {
-                **parent,
-                "children": [],  # Drop the module place holder
-            }
-            # Initialize the data fields.
-            data_field_initialize_func(nexus_group_store[path], data_piece)
-
-        # Merge data piece
-        merge_func(nexus_group_store[path], data_piece)
+    for path in find_insert_paths(structure, data):
+        if path not in nexus_store:
+            nexus_store[path] = initialize(structure, path, data)
+        merge_message(cast(DatasetOrGroup, nexus_store[path]), data)
 
 
-def merge_message_into_nexus_group_store(
-    *,
```

Comment:
We typically make functions keyword-only if it has more than 1-2 arguments.
I think this one especially should be keyword-only, since `structure` and `nexus_store` and `data` can all be dictionaries.